datalad/distribution/add.py

```diff
@@ -404,7 +437,10 @@ def __call__(
                     # in the dataset
                     ap = {k: v for k, v in res.items() if k != 'status'}
                     ap['staged'] = True
-                    to_save.append({k: v for k, v in res.items() if k != 'status'})
+                    # strip any status and state info (e.g. save will refuse to save
+                    # stuff that is marked state='untracked'
```

Comment:
To be honest, I don't entirely get it. If it's marked 'untracked' why not change this, now that it was added, but pretend we have no idea of the state at all?